### PR TITLE
DROOLS-6762 DMN 1.4 support (engine side support)

### PR DESCRIPTION
**Pull request by tarilabs**
Most of this "elbow grease" work for supporting DMNv1.4 in the engine.

**JIRA**:  https://issues.redhat.com/browse/DROOLS-6762

**referenced Pull Requests**: none

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-dmn/kie-dmn-validation/src/test/resources/logback.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
     </encoder>
   </appender>
 
-  <logger name="org.kie" level="warn"/>
+  <logger name="org.kie" level="debug"/>
```
Comment (danielezonca) [inline kie-dmn/kie-dmn-validation/src/test/resources/logback.xml]: Back to `warn` to avoid to pollute CI logs?

**Review by danielezonca — APPROVED** [review]

File: kie-dmn/kie-dmn-validation/src/test/resources/logback.xml (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-validation/src/test/resources/logback.xml]: good catch thanks

Comment (tarilabs) [reply kie-dmn/kie-dmn-validation/src/test/resources/logback.xml]: done with 
eba1821

**Comment by tarilabs** [comment]
Jenkins retest this

**Comment by tarilabs** [comment]
```
kiegroup_kogito_examples/kogito-examples/process-outbox-mongodb-quarkus/src/test/java/org/kie/kogito/examples/OutboxIT.java:134 | org.awaitility.core.ConditionTimeoutException:  Assertion condition defined as a lambda expression in org.kie.kogito.examples.OutboxIT 1 expectation failed.
```
in `Kogito Examples Build / Maven Build (ubuntu-latest, 11, 3.8.1) (pull_request) Failing after 86m`
unrelated ☝️

**Review by hellowdan — APPROVED** [review]

**Pull request merged**